libs/ledger/src/chain/block.cpp

```diff
@@ -76,5 +82,14 @@ void Block::UpdateDigest()
   proof.SetHeader(body.hash);
 }
 
+void Block::UpdateTimestamp()
+{
+  if (body.previous_hash != GENESIS_DIGEST)
+  {
+    const auto ms_since_epoch = duration_cast<milliseconds>(system_clock::now().time_since_epoch());
```

Comment:
Aren't Unix timestamps are seconds since epoch?

I am not sure `system_clock` is mandated to use UTC until C++20. Another approach would be to use `std::time()` + `std::gmtime()` + `std::mktime()` etc.

What are your thoughts?

Reply:
They are seconds, but I thought we'd like millisecond precision on this field.

You're right, it's not mandated. I'll switch to the `<cstdlib>` functions to be safe.